Example/WalletApp/PresentationLayer/Wallet/PushMessages/PushMessagesPresenter.swift

```diff
@@ -40,9 +41,28 @@ extension PushMessagesPresenter: SceneViewModel {
 private extension PushMessagesPresenter {
 
     func reloadPushMessages() {
-        self.pushMessages = interactor.getPushMessages().map({ pushMessageRecord in
-            PushMessageViewModel(pushMessageRecord: pushMessageRecord)
-        })
+        self.pushMessages = interactor.getPushMessages()
+            .sorted {
+                // Most recent first
+                $0.publishedAt > $1.publishedAt
+            }
+            .map { pushMessageRecord in
+                PushMessageViewModel(pushMessageRecord: pushMessageRecord)
+            }
+        
+        interactor.pushMessagePublisher
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] newPushMessage in
+                let newMessageViewModel = PushMessageViewModel(pushMessageRecord: newPushMessage)
+                guard let index = self?.pushMessages.firstIndex(
```

Comment:
may be just update pushMessages without sorting: 
`self.pushMessages = interactor.getPushMessages()`

and create computed property in Presenter: 
```
var viewModels: [PushMessageViewModel] {
return pushMessages
.sort { ... }
.map { PushMessageViewModel(...) }
}
```

Reply:
I would agree if there were more places where we mutate the `pushMessages` array, but we have only a single place atm; also, it's good to be cautious with computed properties used in SwiftUI Views as they can be recomputed quite often. Again it's not the problem in this case, but more like personal preference. WDYT?